test/option_parser.c

```diff
@@ -59,6 +61,39 @@ TEST test_ini_get_string(void)
         PASS();
 }
 
+TEST test_ini_get_path(void)
+{
+        char *section = "path";
+        char *ptr;
+        char *path;
+        wordexp_t replaced_path;
+
+        ASSERT_STR_EQ("/home/test/.path", (ptr = ini_get_path(section, "expand_none", "")));
+        free(ptr);
+
+        path = ini_get_string(section, "expand_home", "");
+        wordexp(path, &replaced_path, WRDE_NOCMD);
```

Comment:
Since we know the contents of the expansion we should use the simplest method (read: `getenv`) to get the data we need to compile the test case. Using the exact same code the function you're testing is using is error prone and can miss quite a few bugs.

 (hint: You can use the `string_replace` function from `utils.c` to do the actual replacement)